CRYOSMSApp/src/CRYOSMSDriver.cpp

```diff
@@ -58,15 +77,294 @@ asynStatus CRYOSMSDriver::writeInt32(asynUser *pasynUser, epicsInt32 value)
 
+	if (envVarMap.at("HEATER_OUT") != NULL) {
+		epicsFloat64 heatOut = std::stod(envVarMap.at("HEATER_OUT"));
+		RETURN_IF_ASYNERROR(putDb, "HEATER:VOLT:_SP", &heatOut);
+	}
+	return status;
+}
 
+asynStatus CRYOSMSDriver::checkUseMagnetTemp()
+{
+	asynStatus status = asynSuccess;
+	int trueVal = 1;
+
+	if (!std::strcmp(envVarMap.at("USE_MAGNET_TEMP"),  "Yes") && (envVarMap.at("MAGNET_TEMP_PV") == NULL || envVarMap.at("MAX_MAGNET_TEMP") == NULL || envVarMap.at("MIN_MAGNET_TEMP") == NULL)) {
+
+		const char *statMsg = "Missing parameters to allow the magnet temperature to be used";
+		this->writeDisabled = TRUE;
+		RETURN_IF_ASYNERROR(putDb, "STAT", &statMsg);
+		RETURN_IF_ASYNERROR(putDb, "DISABLE", &trueVal);
+	}
+	return status;
+}
+
+asynStatus CRYOSMSDriver::checkCompOffAct()
+{
+	asynStatus status = asynSuccess;
+	int trueVal = 1;
+	if (!std::strcmp(envVarMap.at("COMP_OFF_ACT"), "Yes") && (envVarMap.at("NO_OF_COMP") == NULL || envVarMap.at("MIN_NO_OF_COMP_ON") == NULL || envVarMap.at("COPM_1_STAT_PV") == NULL ||
+		envVarMap.at("COMP_2_STAT_PV") == NULL)) {
+
+		const char *statMsg = "Missing parameters to allow actions on the state of the compressors";
+		this->writeDisabled = TRUE;
+		RETURN_IF_ASYNERROR(putDb, "STAT", &statMsg);
+		RETURN_IF_ASYNERROR(putDb, "DISABLE", &trueVal);
+	}
+	return status;
+}
+
+asynStatus CRYOSMSDriver::checkRampFile()
+{
+	asynStatus status;
+	int trueVal = 1;
+	if (envVarMap.at("RAMP_FILE") == NULL) {
+		const char *statMsg = "Missing ramp file path";
+		this->writeDisabled = TRUE;
+		RETURN_IF_ASYNERROR(putDb, "STAT", &statMsg);
+		RETURN_IF_ASYNERROR(putDb, "DISABLE", &trueVal);
+	}
+	else {
+		status = readFile(envVarMap.at("RAMP_FILE"));
+		if (status != asynSuccess) {
+			this->writeDisabled = TRUE;
+			return status;
+		}
+	}
+	double currT;
+	double initRate;
+	int i;
+	RETURN_IF_ASYNERROR(getDb, "OUTPUT:FIELD:TESLA", &currT);
+	for (i = 0; i <= sizeof(pMaxT_); i++) {
+		if (pMaxT_[i] > currT) {
+			break;
+		}
+	}
+	if (i == sizeof(pMaxT_)) {
+		initRate = 0.0;
+	}
+	else {
+		initRate = pRate_[i];
+	}
+	RETURN_IF_ASYNERROR(putDb, "RAMP:RATE:_SP", &initRate);
+	return status;
+}
+
+asynStatus CRYOSMSDriver::onStart()
+{
+	asynStatus status;
+	int trueVal = 1;
+	int falseVal = 0;
+	std::vector<std::string> envVarsNames = {
+		"T_TO_A", "WRITE_UNIT", "DISPLAY_UNIT", "MAX_CURR", "MAX_VOLT", "ALLOW_PERSIST", "FAST_FILTER_VALUE", "FILTER_VALUE", "NPP", "FAST_PERSISTANT_SETTLETIME", "PERSISTNENT_SETTLETIME",
+		"FASTRATE", "USE_SWITCH", "SWITCH_TEMP_PV", "SWITCH_HIGH", "SWITCH_LOW", "SWITCH_STABLE_NUMBER", "HEATER_TOLERANCE", "SWITCH_TOLERANCE", "SWITCH_TEMP_TOLERANCE", "HEATER_OUT",
+		"USE_MAGNET_TEMP", "MAGNET_TEMP_PV", "MAX_MAGNET_TEMP", "MIN_MAGNET_TEMP", "COMP_OFF_ACT", "NO_OF_COMP", "MIN_NO_OF_COMP_ON", "COMP_1_STAT_PV", "COMP_2_STAT_PV", "RAMP_FILE" };
+	for (std::string envVar : envVarsNames)
+	{
+		envVarMap.insert(std::pair<std::string, const char* >(envVar, std::getenv(envVar.c_str())));
```

Comment:
👍 This is a really good way of ensuring that all the environment variables are available here and can be modified by the tests